### ISSUE TITLE
Token seems to be lost after refresh

### ISSUE THREAD

**Issue by wayne-o** [fenichelar/ember-simple-auth-token#46]
Hi

I have all this running using tokens but when the page is refreshed the token is lost from the headers. I can see that it is still in session looking at the local storage.

Problem is that I had to manually add it in the 1st place by overriding the authenticate method on my login controller:

```
authenticate: function() {
            var _this = this;
            this._super()
                .then(function(cb1, cb2) {
                    var user = _this.get('session.user');
                    var token = _this.get('session.access_token') || _this.get("session.token");
                    if (_this.get('session.isAuthenticated') && !Ember.isEmpty(token)) {
                        Ember.$.ajaxSetup({
                            beforeSend: function(xhr) {
                                xhr.setRequestHeader('access_token', token);
                                                                xhr.setRequestHeader('token', token);
                            }
                        });
                    }

                    if (user.username === undefined || user.username == null) {
                        _this.transitionToRoute('preferences');
                    }
                });
        }
```

I'm guessing that this hack should be unneeded and I am doing something wrong elsewhere?

I can see that the `restore` method is called in the token authenticator but yeah - no headers....

**Comment by wayne-o** [comment]
I think this is because the only place we are setting the request headers in JQ is here: https://github.com/jpadilla/ember-cli-simple-auth-token/blob/4b9741a04a0a727f59987434139cf7570592377f/addon/authorizers/token.js#L86

Surely this needs to happen in the restore function too?

**Comment by wayne-o** [comment]
OK so I think that the request is only ever authorized and decorated with access_token when the route is an authenticated one? This doesn't really make sense because you might want to personalise content and the only way I can do that is if the access token is present.

Is this a simple-auth core related debate or a simple-auth-token one?

**Comment by jpadilla** [comment]
Hmmm I've never really tried outside of an authenticated route. [Here](https://github.com/simplabs/ember-simple-auth/blob/master/packages/ember-simple-auth/lib/simple-auth/setup.js) is where the actual authorization is called. None of the route mixins as far as I can see would affect.

**Comment by erichaus** [comment]
@wayne-o this may be related to https://github.com/jpadilla/ember-cli-simple-auth-token/pull/44 which was preventing restore from working after page refresh...

**Comment by leejt489** [comment]
@wayne-o so your token remains in local storage after refresh?  I have a similar issue, but my local storage record is deleted after every refresh.

My environment.js config looks like this

``` javascript
    ENV['simple-auth'] = {
            authorizer: 'simple-auth-authorizer:token',
            store: 'simple-auth-session-store:local-storage'
    };

    ENV['simple-auth-token'] = {
            serverTokenEndpoint: 'http://localhost:1337/login',
            identificationField: 'email',
            passwordField: 'password',
            tokenPropertyName: 'token',
            authorizationPrefix: 'Bearer ',
            authorizationHeaderName: 'Authorization',
            headers: {},
            refreshAccessTokens: false,
            tokenExpireName: 'expires',
            refreshLeeway: 0,
            timeFactor: 1  // example - set to "1000" to convert incoming seconds to milliseconds.
    };
```

and my server returns this format:

``` javascript
{
"token": "...",
"expires": 1430956578803,
"email": "email@domain.com",
"user": {...},
}
```

**Comment by alkamin** [comment]
@leejt489 Same issue here.  Local storage record is removed and all that remains is `{"secure":[]}`

**Comment by alkamin** [comment]
Seems like the problem for me was that the `serverTokenRefreshEndpoint` was not set properly in the environment config

**Comment by erichaus** [comment]
@alkamin is that a bug in the repo? lmk so we can update it if so

**Comment by leejt489** [comment]
@alkamin what is the fix?

**Comment by alkamin** [comment]
@erichonkanen Nope, that was just a setting that is mentioned in a spot I didn't read (it's currently in the customizing section).  I think it could be more prominent as in most cases I feel it will need to be set when using JWT (unless `/api-token-refresh/` is some assumed standard I'm unaware of).

@leejt489 The fix for me was (in my `config/environment.js` file), adding `serverTokenRefreshEndpoint` with the proper URL instead of the default (`/api-token-refresh/`) to the `ENV['simple-auth-token']` object.

**Comment by momer** [comment]
FYI @jpadilla @leejt489 upgrading from 0.6.0 to 0.7.1 resolved this for me on Ember CLI 0.2.5. 

Thank you much for your work on this; consider pushing it up to npm soon; I'm currently using an entry like this in my `package.json`:

```
    "ember-cli-simple-auth-token": "git://github.com/jpadilla/ember-cli-simple-auth-token.git"
```

**Comment by jpadilla** [comment]
@momer great to hear! I'll hopefully publish on npm during the weekend.

**Comment by jpadilla** [comment]
@momer just published v0.7.1. Closing this for now. @wayne-o please give it a try and update this issue if you're still having issues.
